contract/src/lib.rs

```diff
@@ -47,15 +46,49 @@ pub enum ProtocolContractState {
 }
 
 #[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub enum VersionedMpcContract {
+    V0(MpcContract),
+}
+
+impl Default for VersionedMpcContract {
+    fn default() -> Self {
+        VersionedMpcContract::V0(MpcContract::default())
+    }
+}
```

Comment:
hmm, why do we have a `Default` impl? `MpcContract::default()` call will panic because we made it so with the `PanicOnDefault` attribute. We shouldn't have a default impl for `VersionedMpcContract` then since that would give away to calling into an `init` function and overwrite contract state without our control

Reply:
We need to have Default trait implemented because `#[near_bindgen]` requires Default trait. The example [here](https://github.com/near/near-sdk-rs/blob/9aa5eb0b7865e76ac7e89b7cbf46e268be4b56e2/examples/versioned/src/lib.rs) uses #[near(contract_state)] which is not included in the version of near-sdk we are using.
I could make the default function directly panic instead.  `PanicOnDefault` does not work for enums

Reply:
I guess the `PanicOnDefault` was not considered very well for enums. Let's just have our `Default` impl panic then